Examples/GenerativeAICLI/Sources/GenerateContent.swift

```diff
@@ -70,7 +76,32 @@ struct GenerateContent: AsyncParsableCommand {
         name: modelNameOrDefault(),
         apiKey: apiKey,
         generationConfig: config,
-        safetySettings: safetySettings
+        safetySettings: safetySettings,
+        tools: [Tool(functionDeclarations: [
+          FunctionDeclaration(
```

Comment:
As mentioned on chat, I feel that this way of specifying a function is pretty verbose. I'd like us to investigate more Swfit-y ways to implement this, for example:

1) callAsFunction - [https://developer.apple.com/documentation/swiftui/dismissaction/callasfunction()](https://developer.apple.com/documentation/swiftui/dismissaction/callasfunction%28%29)
2) Mark existing Swift functions using a macro
3) Use result builders - https://www.swiftbysundell.com/articles/deep-dive-into-swift-function-builders/

Reply:
Will look into this further in follow-ups.